src/components/Layout/TableOfContents.jsx

```diff
@@ -1,165 +1,90 @@
-import React from "react"
-import Link from 'gatsby-link'
-import styled from 'styled-components'
-
-
-class TableOfContents extends React.Component {
-  constructor() {
-    super();
-    this.nodeListItemsToRender = [];
-    this.currentLevel = 0;
-  }
-
-  formatChapterTitle(title) {
-    return title.split('_').map(word => word[0].toUpperCase() + word.substring(1)).join(' ');
-  }
-
-  addSubchapterJSX(node) {
```

Comment:
State mutation is difficult to read, to debug and to understand. As you can see the same result can be achieved with just evaluating one (big and composed) function without ever mutating any variable.